File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res = 0;
        for(int i = 0; i < s.size(); i++) {
            unordered_set<char> se;
            for(int j = i; j < s.size(); j++) {
                if(se.find(s[j]) != se.end()) break;
                se.insert(s[j]);
            }
            res = max(res, (int)(se.size()));
        }
        return res;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-8.java
class Solution {
    public int maxProfit(int[] prices) {
        int L = 0, R = 0, res = 0;
        while(R < prices.length) {
            if(prices[L] < prices[R]) {
                res = Math.max(res, prices[R] - prices[L]);
            }
            else {
                L = R;
            }
            ++R;
        }
        return res;
    }
}
